ansible/ssh_session_gen.py

```diff
@@ -0,0 +1,193 @@
+            print(
+                """Skip generating SSH session for testbed node: Testbed = {}, Type = {}, Node = {}
+                (SSH IP is not specified)""".format(
+                    testbed.conf_name, testbed_node_type, testbed_node.name
+                )
+            )
+            return
+
+        print(
+            "Start generating SSH session for testbed node: Testbed = {}, Type = {}, Node = {}".format(
+                testbed.conf_name, testbed_node_type, testbed_node.name
+            )
+        )
+
+        session_path = os.path.join(
+            testbed.inv_name,
+            testbed.conf_name,
+            testbed_node_type + "-" + testbed_node.name,
+        )
+        self.repo_generator.generate(
+            session_path,
+            testbed_node.ssh_ip,
+            testbed_node.ssh_user,
+            testbed_node.ssh_pass,
+        )
+
+
+def main(args):
+    print("Loading ansible host inventory: {}\n".format(args.inventory_file_paths))
+    ansible_hosts = HostManager(args.inventory_file_paths)
+
+    print(
+        "Loading testbed config: TestBedFile = {}, Pattern = {}".format(
+            args.testbed_file_path, args.testbed_pattern
+        )
+    )
+    testbeds = TestBed.from_file(
+        args.testbed_file_path, args.testbed_pattern, ansible_hosts
+    )
+    if len(testbeds) == 0:
+        print("No testbeds loaded. Exit.")
+        return
+    else:
+        print("{} testbeds loaded.\n".format(len(testbeds)))
+
+    print(
+        "Starting SSH session repo generation with config: OutputDir = {}, Template = {}".format(
+            args.target, args.template_file_path
+        )
+    )
+    repo_generator = SecureCRTSshSessionRepoGenerator(
+        args.target, args.template_file_path
+    )
+    testbed_repo_generator = TestBedSshSessionRepoGenerator(testbeds, repo_generator)
+    testbed_repo_generator.generate()
+
+
+if __name__ == "__main__":
+    # Parse arguments
+    example_text = """Examples:
+- python3 ssh_session_gen.py -i inventory -o /data/sessions/testbeds -p some_securecrt_session.ini
+- python3 ssh_session_gen.py -i lab t2_lab -n vms-.* -o /data/sessions/testbeds -p some_securecrt_session.ini
+- python3 ssh_session_gen.py -i your_own_inv -t your_own_testbed.yaml -n .*some_tests.* -o /data/sessions/testbeds \
+    -p some_securecrt_session.ini
+
+To generate the SSH session files for SecureCRT, we need to do a few things first:
+1. Install pycryptodome package. We can do it via pip3: pip3 install pycryptodome.
+2. Prepare an existing session file as template (passed by -p). Since our dev machine might not have direct access
+   to your testbed machines, this allows us to inherit the session settings from the template file, such as SSH
+   proxies, fonts and etc. This is te some_securecrt_session.ini file in the examples above.
+3. Setup the `secrets.json` file under `ansible/group_vars/all`. This allows us to get the SSH credentials for the
+   testbed nodes.
```

Comment:
Add instructions for secureCRT to use the generated session repo files?

Reply:
thanks for the advice! got it updated : D